sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -210,14 +210,71 @@ bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
       "PI backend failed. PI backend returns: " + codeToString(Error), Error);
 }
 
+bool handleInvalidWorkItemSize(const device_impl &DeviceImpl,
+                               const NDRDescT &NDRDesc) {
+
+  const plugin &Plugin = DeviceImpl.getPlugin();
+  RT::PiDevice Device = DeviceImpl.getHandleRef();
+
+  size_t MaxWISize[] = {0, 0, 0};
+
+  Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_MAX_WORK_ITEM_SIZE,
+                                          sizeof(MaxWISize), &MaxWISize,
+                                          nullptr);
+  for (int I = 0; I < NDRDesc.Dims; I++) {
+    if (NDRDesc.LocalSize[I] > MaxWISize[I])
+      throw sycl::nd_range_error(
+          "Number of work-items in a work-group exceed limit for dimension "
+          "{I}: {NDRDesc.LocalSize[I]} > {MaxWISize[I]}",
+          PI_INVALID_WORK_ITEM_SIZE);
+  }
+  return 0;
+}
+
 bool handleError(pi_result Error, const device_impl &DeviceImpl,
                  pi_kernel Kernel, const NDRDescT &NDRDesc) {
   assert(Error != PI_SUCCESS &&
          "Success is expected to be handled on caller side");
   switch (Error) {
   case PI_INVALID_WORK_GROUP_SIZE:
     return handleInvalidWorkGroupSize(DeviceImpl, Kernel, NDRDesc);
-  // TODO: Handle other error codes
+
+  case PI_INVALID_KERNEL_ARGS:
+    throw sycl::nd_range_error(
+        "The kernel argument values have not been specified "
+        " OR "
+        "a kernel argument declared to be a pointer to a type"
+        " does not point to a named address space",
+        PI_INVALID_KERNEL_ARGS);
```

Comment:
I see that I suggested to handle this error code in #935, but at the same time the second half of this message is pretty confusing for the SYCL user - he doesn't declare arguments for kernel explicitly: this is done automatically by compiler based on labda capture-list or members of function object used as kernel.

However, in SYCL, this error is also triggered sometimes if kernel captured a non-USM pointer on GPU (I saw several examples with `this` pointer), so, I suggest to rewrite the second part of this message to something like "a kernel captured non-USM pointer"